kernel/task/src/lib.rs

```diff
@@ -77,6 +77,26 @@ use x86_64::registers::model_specific::FsBase;
 use preemption::PreemptionGuard;
 use no_drop::NoDrop;
 
+/// Error types for Task module. If you add another field, then also add corresponding change to the Display impl for TaskError. 
+#[derive(Debug)]
+pub enum TaskError {
+    NotFound,
+    AlreadyExists,
+}
+
+impl fmt::Display for TaskError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TaskError::AlreadyExists => {
+                write!(f, "AlreadyExists")
+            },
+            TaskError::NotFound => {
+                write!(f, "NotFound")
+            },
+        }
+    }
+}
+
```

Comment:
1. This should be moved into the `tls_current_task` mod.
2. This should be named something related to accessing/getting the current task, since that's what it's about. It doesn't have anything to do with the rest of the task subsystem at large.